charts/dex/templates/servicemonitor.yaml

```diff
@@ -0,0 +1,31 @@
+{{- if and .Values.servicemonitor.enabled }}
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
+metadata:
+  name: {{ template "dex.fullname" . }}
+{{- if .Values.servicemonitor.namespace }}
```

Comment:
Any reason why the namespace needs to be overridden?

Reply:
Depends on the prometheus-operator configuration, i.e. where it looks for serviceMonitors. Sometimes you want to put the serviceMonitor into the monitoring-namespace in case the operator only looks there. So it should be possible to override the namespace with the release-namespace as the default.